routes/variableurl.go

```diff
@@ -0,0 +1,123 @@
+package routes
+
+import (
+	"net/http"
+	"strconv"
+	"strings"
+)
+
+type (
+	parseIndex uint
+
+	ArgName string
+
+	ParseInt struct {
+		ArgName ArgName
+	}
+	ParseFloat32 struct {
+		ArgName ArgName
+	}
+	ParseFloat64 struct {
+		ArgName ArgName
+	}
+	ParseString struct {
+		ArgName ArgName
+	}
+	ParseVoid struct {
+		Required string
+	}
+	ParseEnd struct{}
+)
+
+const (
+	index = parseIndex(iota)
+)
+
+var (
+	badURLTypeMessage  = ErrorBody{"Bad URL type."}
```

Comment:
When you send this, the URL is fine : the type of the argument is unexpected. The name and message should reflect that.